src/util/sighash.rs

```diff
@@ -87,6 +89,29 @@ pub struct ScriptPath<'s> {
     leaf_version: u8,
 }
 
+/// Hashtype of an input's signature, encoded in the last byte of the signature
+/// Fixed values so they can be casted as integer types for encoding
+#[derive(PartialEq, Eq, Debug, Copy, Clone)]
+pub enum SigHashType {
+    /// 0x0: Used when not explicitly specified, defaulting to [SigHashType::All]
+    Default = 0x00,
+    /// 0x1: Sign all outputs
+    All = 0x01,
+    /// 0x2: Sign no outputs --- anyone can choose the destination
+    None = 0x02,
+    /// 0x3: Sign the output whose index matches this input's index. If none exists,
+    /// sign the hash `0000000000000000000000000000000000000000000000000000000000000001`.
+    /// (This rule is probably an unintentional C++ism, but it's consensus so we have
+    /// to follow it.)
+    Single = 0x03,
+    /// 0x81: Sign all outputs but only this input
+    AllPlusAnyoneCanPay = 0x81,
```

Comment:
Plus is a bit of a weird word since it's a logical AND of the functionality (bitwise OR is plus though).

Reply:
For better or for worse, this is also the current API in the already existing SigHashType. IMO it's better to be consistent.